src/main/java/kitchenpos/application/MenuService.java

```diff
@@ -40,53 +41,69 @@ public MenuService(
 
     @Transactional
     public Menu create(final MenuCreateRequest menuCreateRequest) {
-        final Long price = menuCreateRequest.getPrice();
+        MenuProducts menuProducts = generateMenuProducts(menuCreateRequest);
+        Menu menu = menuDao.save(generateMenu(menuCreateRequest, menuProducts));
 
-        if (Objects.isNull(price) || price < 0) {
-            throw new IllegalArgumentException();
+        final Long menuId = menu.getId();
+        List<MenuProduct> results = menuProducts.changeMenuId(menuId).getMenuProducts();
+        results.forEach(menuProductDao::save);
+        return menu.changeMenuProducts(results);
+    }
+
+    public List<Menu> list() {
```

Comment:
```suggestion
    public List<Menu> list() {
      return menuDao.findAll()
                .stream()
                .peek(menu -> menu.setMenuProducts(menuProductDao.findAllByMenuId(menu.getId())))
                .collect(Collectors.toList());
    }
```
이건 어때요? 추가로 setter를 사용하지 않고 다른 방법을 사용해 peek을 map으로바꿀 수 있게 하면 더 좋을거 같아요!

Reply:
해당 코드가 DTO 작업을 하면서 꽤 변경이 되었으니 다시 봐주시면 감사하겠습니다. 추가로 제 코드에는 별도의 setter는 없고 해당 기능을 담당하는 메소드는 전부 새로운 객체를 반환하도록 구현하였습니다